website/src/docs/concepts.mdx

```diff
@@ -4,5 +4,119 @@ export const info = {
+The other is the *empty* one with no networking, no files, no environment
+variables. Sandboxes tend to approximate this environment. Checks and builds
+run in a combination of this empty environment and the environment you specify
+in the check.
+
+If `myEnv` has type `Environment`, you get two methods on `myEnv` that relate
+to these aforementioned environments:
+
+- `myEnv.shell`: This is a template literal that runs the provided command in
+  the combination of `myEnv` and the current environment. This is useful for
+  development commands.
+- `myEnv.check`: This is a template literal that runs the provided command in
+  the combination of `myEnv` and the empty environment. Because it doesn't allow
+  access to the current environment, it is reproducible: if you run it twice,
+  the result will be the same<Hovernote>So much so that, as an optimization,
+  `garn` and `nix` store the result of the first run to avoid rerunning a second
+  time.</Hovernote>. And if you run it on different computers, then, with some
+  caveats, the result will be the same<Hovernote>So much so that, as an optimization,
+  if you run it a CI like <a href="https://garnix.io">garnix</a> and set it up
+  as a cache, and then run it locally, the result obtained by **garnix** will
+  be downloaded and reused, rather than run locally</Hovernote>
+
+
+
+## Package
+
+If you run an executable, it may create new files. If you run the executable
+in an environment without access to the current environment - in a sandbox, if
+you will - *and* you designate a particular set of files as "worth keeping" by
+moving them into a special directory, we say that you are *building* a *package*.
+The package then contains all those files moved into the special directory.
+
+That's very abstract, so let's make it more concrete. If you have some **go**
+source code, you could compile that source code into an executable. You now
+want to keep that executable, so you can use it in various ways. So you create
+a package; the build process of that package is compiling that executable, and
+marking it as special by moving it into an `$out` directory. The *contents* of
+the package are then just the executable.
+
+You build all packages in a project with "`garn build <PROJECT>`"
+
+## Check
+
+A *check* is quite similar to a package, in that it also runs a command in a
+sandbox-like environment. But here we *do not* care about built files, but *only*
+the exit code<Hovernote>And the stdout and stderr of the process for debugging the check.</Hovernote>.
+
+You can run all checks in a project with "`garn check <PROJECT>`"
+
+## Executable
+
+An executable is a thing that can be run, without reference to the context in
+which it is run. In practice, they are run in the current environment via the
+"garn run" command. This provides a very easy way of making tools available
+to collaborators in your project.
+
+Usually executables will be a particular file in a package.
```

Comment:
Our `Executables` are often commands, including command line arguments. Under the hood they're translated into an executable file in a nix package, but that is not exposed through something of type `Package`.

Reply:
Yeah, maybe my definition isn’t great. How would you define an Executable? Just like Build/Check, but always running in an augmentation of the current environment?

Reply:
Yeah, something like that. Here's my attempt at a concepts page: #258. So that's how I would explain all the concepts. (I thought it was easiest to just start from scratch in another draft PR. Not saying we should abandon this one.)